stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -1937,8 +1936,8 @@ public class ChatClient internal constructor(
 
         @InternalStreamChatApi
         @ExperimentalStreamChatApi
-        public fun withPlugin(plugin: Plugin): Builder = apply {
-            plugins += plugin
+        public fun withPlugin(plugin: PluginFactory): Builder = apply {
```

Comment:
Factory instead of the actual implementation, this way the initialization of the Plugin belongs to the factory, and then the plugin is not aware of its creation/initialization.